report_templates/host_compare_content_hosts_packages.erb

```diff
@@ -0,0 +1,70 @@
+<%#
+name: Host - compare content hosts packages
+snippet: false
+template_inputs:
+- name: Host 1
+  required: true
+  input_type: user
+  advanced: false
+  value_type: plain
+  resource_type: Katello::ActivationKey
+- name: Host 2
+  required: true
+  input_type: user
+  advanced: false
+  value_type: plain
+  resource_type: Katello::ActivationKey
+model: ReportTemplate
+-%>
+<%-  general_lst = [] -%>
+<%-  stage_lst = [] -%>
+<%-  final_lst = [] -%>
+<%-  arch_lst = [] -%>
+<%-  host1 = Host.find_by_name(input("Host 1")) -%>
+<%-  host2 = Host.find_by_name(input("Host 2")) -%>
+<%-  set1 = host1.installed_packages.group_by { |p| p.name } -%>
+<%-  set2 = host2.installed_packages.group_by { |p| p.name } -%>
+<%-  (set1.keys | set2.keys).uniq.each do |package_name| -%>
+<%-    data = { 'Package' => package_name } -%>
+<%-    data[host1.fqdn] = set1[package_name].try(:first).try(:nvra) || '-' -%>
+<%-    data[host2.fqdn] = set2[package_name].try(:first).try(:nvra) || '-' -%>
+<%-    general_lst << data -%>
+<%-  end -%>
+<%-  general_lst.each do |d| -%>
+<%-    ch1_ver = d[host1.fqdn].gsub(d["Package"]+"-","").gsub(/\.el[0-9][^.]*/,"").gsub(/\.[n|x|i].*/,"") -%>
+<%-    ch2_ver = d[host2.fqdn].gsub(d["Package"]+"-","").gsub(/\.el[0-9][^.]*/,"").gsub(/\.[n|x|i].*/,"") -%>
+<%-    arch_lst << d[host1.fqdn].split(/\./).last -%>
+<%-    arch_lst << d[host2.fqdn].split(/\./).last -%>
+<%-    arch_lst = arch_lst.uniq -%>
+<%-    arch_lst.delete("-") -%>
+<%-    if ch1_ver == "-" -%>
+<%-      status = "#{host2.fqdn} only"-%>
+<%-    elsif ch2_ver == "-" -%>
+<%-      status = "#{host1.fqdn} only"-%>
+<%-    elsif ch1_ver > ch2_ver -%>
```

Comment:
Does this compare versions properly? For example, `3.10 > 3.2` in versions but in Ruby that won't be true. While within RHEL it may remain the same, you will hit it with releases as well.

Reply:
Yeah, we may need to wrap it with Gem version comparison, probably deserves a macro

Reply:
this is still an open item

Reply:
addressed  in the last commit in this PR